clippy_lints/src/methods/str_splitn.rs

```diff
@@ -209,3 +242,120 @@ fn parse_iter_usage(
         span,
     })
 }
+
+use super::NEEDLESS_SPLITN;
+
+pub(super) fn check_needless_splitn(
+    cx: &LateContext<'_>,
+    method_name: &str,
+    expr: &Expr<'_>,
+    self_arg: &Expr<'_>,
+    pat_arg: &Expr<'_>,
+    count: u128,
+) {
+    if !cx.typeck_results().expr_ty_adjusted(self_arg).peel_refs().is_str() {
+        return;
+    }
+    let ctxt = expr.span.ctxt();
+    let mut app = Applicability::MachineApplicable;
+    if method_name == "splitn" && count >= 2 && check_iter(cx, ctxt, cx.tcx.hir().parent_iter(expr.hir_id), count) {
+        span_lint_and_sugg(
+            cx,
+            NEEDLESS_SPLITN,
+            expr.span,
+            "unnecessary use of `splitn`",
+            "try this",
+            format!(
+                "{}.{}({})",
+                snippet_with_context(cx, self_arg.span, ctxt, "..", &mut app).0,
+                "split",
+                snippet_with_context(cx, pat_arg.span, ctxt, "..", &mut app).0
+            ),
+            app,
+        );
+        return;
+    }
+
+    if_chain! {
+        if let ExprKind::Lit(ref self_lit) = self_arg.kind;
+        if let LitKind::Str(self_str, _) = self_lit.node;
+        if let ExprKind::Lit(ref pat_lit)= pat_arg.kind;
+        if let Some(pat) = match pat_lit.node {
+            LitKind::Str(pat_str, StrStyle::Cooked) => {
+                Some(pat_str.to_string())
+            },
+            LitKind::Char(pat_char) => {
+                Some(pat_char.to_string())
+            },
+            _ => {
+                None
+           },
+        };
+        then {
+            let (method_sugg, msg) = if method_name == "splitn" {
+                ("split", "unnecessary use of `splitn`")
+            } else {
+                ("rsplit", "unnecessary use of `rsplitn`")
+            };
+            let self_str = self_str.as_str();
+            let v: Vec<&str> = self_str.split(pat.as_str()).collect();
+            if count >= v.len() as u128 {
+                span_lint_and_sugg(
+                    cx,
+                    NEEDLESS_SPLITN,
+                    expr.span,
+                    msg,
+                    "try this",
+                    format!(
+                    "{}.{}({})",
+                    snippet_with_context(cx, self_arg.span, ctxt, "..", &mut app).0,
+                    method_sugg,
+                    snippet_with_context(cx, pat_arg.span, ctxt, "..", &mut app).0
+                    ),
+                    app);
```

Comment:
Using `splitn` on a string literal seems really unrealistic to me. Is it even worth it for Clippy to detect this? Suggesting to write `"a,b".split(',')` seems questionable since they probably shouldn't use `split` at all. (could write `["a", "b"]`)

Reply:
> Using `splitn` on a string literal seems really unrealistic to me. Is it even worth it for Clippy to detect this? Suggesting to write `"a,b".split(',')` seems questionable since they probably shouldn't use `split` at all. (could write `["a", "b"]`)

Hi，I'm sorry, I have more things to deal with than I thought. I guess I can modify this PR according to your suggestions during the weekend. Thank you very much.

Reply:
To be clear I'm suggesting to not detect this case. Maybe we can add it in a different PR.